src/main/scala/za/co/absa/spark/commons/schema/SchemaUtils.scala

```diff
@@ -17,13 +17,41 @@
 package za.co.absa.spark.commons.schema
 
 import org.apache.spark.sql.functions.{col, struct}
-import org.apache.spark.sql.types.{ArrayType, DataType, StructField, StructType}
+import org.apache.spark.sql.types._
 import org.apache.spark.sql.{Column, DataFrame}
 import za.co.absa.spark.commons.adapters.HofsAdapter
 
-import scala.util.Random
+import scala.annotation.tailrec
 
 object SchemaUtils extends HofsAdapter {
+
+  /**
+   * Returns the parent path of a field. Returns an empty string if a root level field name is provided.
+   *
+   * @param columnName A fully qualified column name
+   * @return The parent column name or an empty string if the input column is a root level column
+   */
+  def getParentPath(columnName: String): String = {
+    val index = columnName.lastIndexOf('.')
+    if (index > 0) {
+      columnName.substring(0, index)
+    } else {
+      ""
+    }
+  }
+
+  /**
+   * Converts a fully qualified field name (including its path, e.g. containing fields) to a unique field name without
+   * dot notation
+   *
+   * @param path the fully qualified field name
+   * @return unique top level field name
+   */
+  def unpath(path: String): String = {
```

Comment:
Questioning its `SparkCommons` status.
This is really weird function. I mean it works in our cases, but it rather utilitarian, not very conceptual, with a cryptic name.